### DISCUSSION TITLE
v0.4.6

### DISCUSSION THREAD

**Discussion by betolink** [earthaccess-dev/earthaccess#185] [Announcements]
This is the first formal release under the new name. 0.4.6 will be available in both pypi and conda-forge. 

The first thing to mention is the new API notation that should evolve to support all the use cases, 

```python
import earthaccess

earthaccess.login(strategy="netrc")

granules = earthaccess.search_data(params)

earthaccess.download(granules, local_path= "./test")
```

is equivalent to

```python
from earthdata import Store, Auth, DataGranules

auth = Auth()
auth.login(strategy="netrc")
store = Store(auth)

granules = DataGranules().params(params).get()

store.get(granules, local_path="./test")

```

We can still use the classes the same way but eventually we should support only module-level API.

Features:

* search datasets by DOI, e.g.
```python
datasets = earthaccess.search_datasets(
    doi="10.5067/AQR50-3Q7CS"
    cloud_hosted=True
)
```
searching by DOI should usually return only one dataset but I'm not sure what would happen if the same data is also in the cloud so to be sure we can use the `cloud_hosted` parameter if we want to operate on the AWS hosted version.

The documentation started to get updated and soon we should have a "gallery" with more examples of how to use the library. 

<hr /><em>This discussion was created from the release <a href='https://github.com/nsidc/earthaccess/releases/tag/v0.4.6'>v0.4.6</a>.</em>

**Comment by betolink** [comment]
There is a bug in the store when we open the files using direct access (S3). This is being fixed and should work tomorrow but I wonder if we can run integration tests in us-west-2 as the data will only be available there.

**Reply by jhkennedy** [reply]
Unfortunately for this, all GitHub.com hosted runners are Azure-based, so  you'd need to either:
* deploy something (lambda/ec2/etc) as part of the action to run it
* or use a self-hosted runner: https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners

The question is always "who funds it" -- I wonder if there's space in OpenScapes, NSIDC/some DAAC, or EDC to do this?

Otherwise, maybe we should write a grant, or build it into a grant, to pilot that?

**Reply by betolink** [reply]
Lambdas are a possibility and NSIDC has some cloud credits that we could use; getting funding and/or applying to an AWS academic program are good ideas tho... Chelle mentioned new funding opportunities at the openscapes slack, maybe we can talk about this a bit more at AGU!
